### DISCUSSION TITLE
Reordering Test Cases

### DISCUSSION THREAD

**Discussion by doctorwu** [Kong/insomnia#3657] [Ideas]
As a user, I would like the ability to reorder the test cases in the Test UI by dragging them around.

**Comment by palevi67** [comment]
Yes, if you are dealing with sessions, execution order is very important!

**Comment by yleflour** [comment]
Old suggestion, but as a newcomer I feel the testing side of things is lacking compared to the incredible debug creation flow. I think the testing side of things really needs the ability to reorder and duplicate to become fully maintainable on the user side

Another way to do this would be to allow for raw editing

**Comment by GreatToCreate** [comment]
Jumping in to also say that test re-ordering would be extremely beneficial. I organize my test suites by process, and when those processes naturally evolve it's unfortunate that I have to re-create the entire suite.

**Comment by PierreStainton** [comment]
+1 !

**Comment by sudo-run-dos-run** [comment]
+1 This is urgently needed if Insomnia wants to be a test tool.

**Comment by wonson** [comment]
+1

Any raw text file editing can be as a workaround for now?

**Comment by Tignus** [comment]
+1

**Comment by palevi67** [comment]
Because of this, I've abandoned insommia in favor of httpyac vscode plugin

**Comment by nathanbottomley** [comment]
+1

**Comment by Ayrtonalec** [comment]
How isn't there even replied to this yet
+1

**Comment by davehouser1** [comment]
Wow I find it quite surprising that this is not a feature. What if I had 50 test cases and I needed to add a new one as the second test case? That requires me to delete the below 49, add the specific test case, and rebuild 49. I know the Kong team is doing a lot but this seems like it should have been implemented a while ago. Any update on this?

**UPDATE (on my end)**

I found a workaround for now (Warning: this will double your data up)
1. Preferences > Export data (I did as JSON)
1. Open the json in your fav text editor ( I recommend using notepad++ with the plugin JSTool. Once installed run JSFormat to pretty adjust the backup file)
1. Now scroll to the bottom, this is where all tests are created. 
1. Adjust the "created" value to integer based time values that fit in between, before, or after the tests you want to move the given test to. Be aware you do not need to re-order the actual JSON objects for each test, just adjust the integers. 
1. If you don't want doubled up data remove everything except your tests. I don't do this, as the format of the parent structure of the json file is not clear to me, so I just import everything again. 
1. Save the file.
1. In insomnia > preferences > data > import, choose file. Scan and then import.
1. Now go to your tests, your unit tests you adjusted should have moved to the locations you wanted them. 
1. Delete duplicates (debugs, tests, and evn vars)

If you want to ADD tests and put them in some order, you need to first, create as many "new tests" as you want. I leave them as the default, but create as many as you need before doing the steps above. Then once you edit the backup, you can see all your new tests. Now you can adjust them to the new created date value. 

This is a nightmare to deal with and a total pain point, mainly because I have found no way to pick and choose what I want to import. Nor have I found a way to have insomnia to import a backup that is not a merge. So I am forced to delete every debug, test, and env var, before trying to import the new data. This is a terrible work around but its the only thing I found that works. Insomnia team please fix this! 
+1

**Comment by elrondurpg** [comment]
The fact that I can't reorder or duplicate test cases in Insomnia is THE reason why I have decided not to use the application as part of my development process right now.

**Comment by hubble459** [comment]
+1
Anyone working on a plugin for this?

**Comment by subnetmarco** [comment]
This has already been built and merged this week: https://github.com/Kong/insomnia/pull/7020

As such, it will be available in the next release.
